File: packages/swift/Sources/CatalystSwift/Models/CatalystCustomReaction.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

// GET /catalyst/v1/reactions が返す標準絵文字リアクション定義
public struct CatalystCustomReaction: Decodable, Sendable, Equatable, Hashable {
  public let id: String
  public let name: String
  public let symbol: String
  public let url: String
}

// GET /catalyst/v1/custom-reactions, POST /catalyst/v1/custom-reactions, PATCH
// /catalyst/v1/custom-reactions/{id} の items に含まれるユーザー独自リアクション
//
// `status` is one of "active" | "moderated" | "hidden" | "disabled".
// `visibility` is one of "private" | "followers" | "public".
public struct CatalystUserCustomReaction: Decodable, Sendable, Equatable, Hashable {
  public let id: String
  public let shortcode: String
  public let displayName: String
  public let imageUrl: String
  public let mimeType: String
  public let sortOrder: Int
  public let status: String
  public let visibility: String
  public let createdAt: String
}

public struct CatalystCustomReactionList: Decodable, Sendable {
  public let plan: String
  public let limit: Int
  public let used: Int
  public let items: [CatalystUserCustomReaction]
}

File: packages/dotnet/CatalystSharp/Models/CatalystCustomReaction.cs
using System.Text.Json.Serialization;

namespace CatalystSharp.Models;

// GET /catalyst/v1/reactions が返す標準絵文字リアクション定義
public record CatalystCustomReaction(
    [property: JsonPropertyName("id")] string Id,
    [property: JsonPropertyName("name")] string Name,
    [property: JsonPropertyName("symbol")] string Symbol,
    [property: JsonPropertyName("url")] string Url
);

// GET /catalyst/v1/custom-reactions, POST /catalyst/v1/custom-reactions, PATCH /catalyst/v1/custom-reactions/{id} の items に含まれるユーザー独自リアクション
// status: "active" | "moderated" | "hidden" | "disabled"
// visibility: "private" | "followers" | "public"
public record CatalystUserCustomReaction(
    [property: JsonPropertyName("id")] string Id,
    [property: JsonPropertyName("shortcode")] string Shortcode,
    [property: JsonPropertyName("displayName")] string DisplayName,
    [property: JsonPropertyName("imageUrl")] string ImageUrl,
    [property: JsonPropertyName("mimeType")] string MimeType,
    [property: JsonPropertyName("sortOrder")] int SortOrder,
    [property: JsonPropertyName("status")] string Status,
    [property: JsonPropertyName("visibility")] string Visibility,
    [property: JsonPropertyName("createdAt")] string CreatedAt
);
